src/ssl.c

```diff
@@ -14124,7 +14234,21 @@ int AddSessionToCache(WOLFSSL_CTX* ctx, WOLFSSL_SESSION* addSession,
             return MEMORY_E;
         }
     }
-#endif
+#if defined(WOLFSSL_TLS13) && defined(WOLFSSL_TICKET_NONCE_MALLOC)
+    if (addSession->ticketNonce.data != addSession->ticketNonce.dataStatic) {
+        preallocNonce = (byte*)XMALLOC(addSession->ticketNonce.len, NULL,
```

Comment:
This one is missing the heap hint. If it is available we need to pass it. This is used for static memory. Also below in the XFREE case.

Reply:
I've some questions regarding this part. 
As the malloc must be done when we don't have the lock on the cache session, we can do that only with `addSession->heap`, but the buffer may end up being in the `cacheSession`. We will have a buffer in `cacheSession` allocated with the heap hint of another session. Is this a problem?
A similar problem can happen if `cacheSession` object had a dynamic buffer that needed to be freed, as it needs to be freed outside the lock, saving `cacheSession->heap` during the lock and using that for doing the free outside the lock, is it safe?
Ticket buffers, that follow a similar pattern of TicketNonce buffers introduced in this PR, use NULL for heap hints.

Reply:
@JacobBarthelmeh do we have any use cases with different heap hints? To me this seems like an edge case we don't support. It should be the same heap hint for both.

Reply:
From my experience using different allocation parameters will result in an error. It needs to be the same.